lib/tapioca/helpers/rbi_files_helper.rb

```diff
@@ -152,49 +152,78 @@ def parse_and_index_files(index, files, number_of_workers:)
       index.visit_all(trees)
     end
 
+    # Do the list of `nodes` sharing the same name have duplicates?
     #: (Array[RBI::Node] nodes, shim_rbi_dir: String, todo_rbi_file: String) -> bool
     def shims_or_todos_have_duplicates?(nodes, shim_rbi_dir:, todo_rbi_file:)
+      # If there is only one node, there are no duplicates
       return false if nodes.size == 1
 
+      # Extract the nodes from the sorbet/rbi/shims/ directory and the todo.rbi file
       shims_or_todos = extract_shims_and_todos(nodes, shim_rbi_dir: shim_rbi_dir, todo_rbi_file: todo_rbi_file)
       return false if shims_or_todos.empty?
 
+      # Extract the nodes that are not shims or todos (basically the nodes from the RBI files generated by Tapioca)
       not_shims_or_todos = nodes - shims_or_todos
-      shims_or_todos_empty_scopes = extract_empty_scopes(shims_or_todos)
 
-      # We need to discard classes that are redefining the parent of a class
-      shims_or_todos_empty_scopes.select! do |scope|
-        # Empty modules are always duplicates
-        next true unless scope.is_a?(RBI::Class)
-
-        # Empty classes without parents are also duplicates
-        parent_name = scope.superclass_name
-        next true unless parent_name
+      # First let's look into scopes (classes, modules, sclass) for duplicates
+      shims_or_todos_scopes = shims_or_todos.grep(RBI::Scope)
+      if shims_or_todos_scopes.any?
```

Comment:
Could we extract helper functions for these, so that this methods can return something like:

```
    has_duplicate_scopes(shims_or_todos)
      || has_duplicate_mixins(shims_or_todos)
      || has_duplicate_methods(shims_or_todos)
```

Reply:
Good idea, done 👍